### PR TITLE
3-2-stable: add ruby 2.2 compatibility

### PR DESCRIPTION
**Pull request by tmm1**
Fixes some warnings that were preventing our rails app from running on ruby 2.2.0-rc1 with rails 3.2.

### CODE REVIEW

**Comment by rafaelfranca** [comment]
Ok. Just created a new branch to run the tests against Ruby 2.1 and 2.2 https://github.com/rails/rails/commit/7c8fb95495bd.

I'm not saying that we will support them yet, all depends in the number of changes required to make it work.

File: activesupport/lib/active_support/values/time_zone.rb [hunk 1]
```diff
@@ -267,7 +267,7 @@ def at(secs)
     #
     #   Time.zone.now                 # => Fri, 31 Dec 1999 14:00:00 HST -10:00
     #   Time.zone.parse('22:30:00')   # => Fri, 31 Dec 1999 22:30:00 HST -10:00
-    def parse(str, now=now)
+    def parse(str, now=now())
```
Comment (korny) [inline activesupport/lib/active_support/values/time_zone.rb]: Wouldn't it be nicer to use `now=self.now`?

Comment (dmitry) [inline activesupport/lib/active_support/values/time_zone.rb]: I agree with that.

**Comment by rafaelfranca** [comment]
Lets see what https://travis-ci.org/rails/rails/builds/45705088 says.

**Comment by rafaelfranca** [comment]
There are still a lot of changes to do to get the build green. I don't have idea how much work we still need and I don't want to take this task, so if anyone care about supporting Ruby 2.2 in Rails 3.2 I recommend to work on top of `rm-3-2-with-ruby-2-1-plus` branch and open a new PR with everything green.

Depending in the amount of changes I can do a new Rails 3.2 release including these changes.

File: activesupport/lib/active_support/values/time_zone.rb (hunk 1 above)
Comment (loganapayne) [inline activesupport/lib/active_support/values/time_zone.rb]: @matz [recommended `foo = foo()`](https://bugs.ruby-lang.org/issues/10314#note-4)

Comment (tonyta) [inline activesupport/lib/active_support/values/time_zone.rb]: @matz [recommended `foo = foo()`](https://bugs.ruby-lang.org/issues/10314#note-4)

Comment (dmitry) [inline activesupport/lib/active_support/values/time_zone.rb]: @tonyta thanks for the reference. I can only see he said about: `To get old behavior, try foo = foo().`

**Comment by bjm88** [comment]
Hi - can someone confirm the latest ruby version rails 3.2.latest will support...Is there a file in github we can check to confirm auto/regression testing wise that its actually covering what versions of ruby...

It sounds like 3.2 on ruby 2.2 is questionable.....how about 3.2 on ruby 2.1, certified?

**Comment by IslamAzab** [comment]
@bjm88 I recommend you check that you check this pull request https://github.com/rails/rails/pull/18306

**Comment by bjm88** [comment]
Thank you IslamAzab.  It seems based off that, 3.2-stable branch is compatible and supported with ruby 2.2, however they did not issue a new release number for 3.2, ie 3.2.22 or something....I guess its fine if we pull from 3.2.stable...

Not sure why they wouldn't put a new 3.2. patch out for such an important change.

**Comment by IslamAzab** [comment]
@rafaelfranca metioned (here)[https://github.com/rails/rails/pull/18306#issuecomment-69333576]
> It will be not any new release of 3-2-stable branch. But if we have a security issue that needed to be fixed on 3.2, we will release from that branch.

File: activesupport/lib/active_support/core_ext/date_time/calculations.rb [hunk 2]
```diff
@@ -138,6 +138,12 @@ def utc_offset
 
   # Layers additional behavior on DateTime#<=> so that Time and ActiveSupport::TimeWithZone instances can be compared with a DateTime
   def <=>(other)
-    super other.kind_of?(Infinity) ? other : other.to_datetime
+    if other.kind_of?(Infinity)
+      super
+    elsif other.respond_to? :to_datetime
+      super other.to_datetime
+    else
```
Comment (andresilveira) [inline activesupport/lib/active_support/core_ext/date_time/calculations.rb]: @egilburg I think here the goal is to explicitly point out the return value. In this case `nil`.

**Comment by rajiv** [comment]
rails 3.2.22 is out and includes ruby 2.2 compatibility.

http://weblog.rubyonrails.org/2015/6/16/Rails-3-2-22-4-1-11-and-4-2-2-have-been-released-and-more/

**Comment by kaspth** [comment]
@rajiv Mind you, that support is not really official and 3.2 is only receiving security releases. You're on your own with it :smile:

**Comment by IslamAzab** [comment]
@kaspth What do you mean by "not really official" ? Wasn't the fix merged to the stable branch and all its tests were passing :confused: ? Can you please clarify that more? Thanks :+1:

**Comment by matthewd** [comment]
@IslamAzab if there are other yet-to-be-found incompatibilities, we won't be making a release to address them

**Comment by dmitry** [comment]
@IslamAzab Tests are passed on Ruby 2.2 (https://github.com/rails/rails/blob/v3.2.22/.travis.yml#L10) but nobody will support it as everyone in the rails core right now are working on Rails *5*.

**Comment by IslamAzab** [comment]
@matthewd @dmitry Thanks for the clarification :)

**Pull request closed without merging**